src/lib/operators/jit_operator/jit_types.hpp

```diff
@@ -6,9 +6,28 @@
 #include "storage/base_value_column.hpp"
 #include "storage/chunk.hpp"
 #include "storage/column_iterables/base_column_iterators.hpp"
+#include "storage/table.hpp"
+#include "types.hpp"
 
 namespace opossum {
 
+// Functions using strings are not optimised to support code specialisation on Linux.
+// This specialiation issue came up with pr #933 (https://github.com/hyrise/hyrise/pull/933).
+// opt none flag to ensure that functions handling strings are not optimized
+#if __has_attribute(optnone)
+#define OPTNONE __attribute__((optnone))
+#else
+#define OPTNONE
+#endif
+// NO_SPECIALISATION flag can be set for debugging, if query is interpreted
+#ifdef NO_SPECIALISATION
+#define JitDebugAssert(expr, msg) DebugAssert(expr, msg)
```

Comment:
It is not necessary. To successfully specialize the operator code I had to remove some `Fail` and `DebugAssert` as these use `std::string`. Making these checks optional was the easiest idea to address the problem. All of them either are already asserted in non-specialized code are can be checked during compilation so that I will remove all of them.